webapp/src/ts/services/enketo.service.ts

```diff
@@ -714,6 +692,24 @@ export class EnketoService {
     form.output.update();
   }
 
+  private setupNavButtons($wrapper, currentIndex) {
+    if(!this.currentForm.pages) {
+      return;
+    }
+    const lastIndex = this.currentForm.pages.activePages.length - 1;
+    const footer = $wrapper.find('.form-footer');
+    footer.removeClass('end');
+    footer.find('.previous-page, .next-page').removeClass('disabled');
+
+    if (currentIndex >= lastIndex) {
+      footer.addClass('end');
+      footer.find('.next-page').addClass('disabled');
+    }
```

Comment:
Sadly no. The `validate` function will actually clear out the values from any non-relevant fields in the form before it actually does the validation.  So, anything that is not relevant before `validate` is just lost.

Unfortunately, Kenn has discovered that this code here is will not work either, though. Apparently, our patch for keeping the `inputs` also somehow still avoided having the `constraint`s checked for the `inputs`. But now that we are actually setting them to relevant here, the constraints are being checked and failing on some forms.  

I am going to look for a different way to keep these `inputs`.

Reply:
Okay, so I got this working again and was able to just get rid of all this code here.  The bad news is that to do this, I had to re-add one of the patches I had removed and also introduce a brand new patch for enketo-core.  :sob: 

There were two main problems that I ran into with the implementation I had here. First, if I set `inputs` to relevant before verification, then any `constraint`s for `inputs` would be evaluated (as mentioned above) and there does not seem to be any clean way to avoid that.  To address this, I had to patch the Enketo `relevant.js` to ensure `inputs` is not cleared during validation (even if it is not relevant).  Second, even with this error handling code where I reset the relevance if the validation fails, this still does not really cover the case where an error gets thrown out of literally anywhere else in this logic (between the time I force the relevance to `true` and the time the form actually get saved...).  Ultimately, I was just not super comfortable actually changing the `relevant` values in the form since it seems too easy to hit unintended side-effects and get dropped back into the form with the `inputs` now visible to the user....  I think it might be possible to find a clean more holistic approach to this, but for now it seems acceptable to just keep the existing patch for the Enketo `form.js` that we have been using for years....